[Writing Tools] Remove unnecessary infrastructure to support concurrent Writing Tools sessions

rr-codes opened the pull request:
#### 4ef46f4e4c9551d4f25abb4d564e16bb0c2fe9d9
<pre>
[Writing Tools] Remove unnecessary infrastructure to support concurrent Writing Tools sessions
<a href="https://bugs.webkit.org/show_bug.cgi?id=277998">https://bugs.webkit.org/show_bug.cgi?id=277998</a>
<a href="https://rdar.apple.com/133726884">rdar://133726884</a>

Reviewed by Aditya Keerthi.

There&apos;s no need to support multiple sessions, and doing so significantly increases the complexity of the code.

* Source/WebCore/dom/DocumentMarker.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebCore::ChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebCore::ChromeClient::removeTransparentMarkersForActiveWritingToolsSession):
(WebCore::ChromeClient::removeInitialTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::addInitialTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::addSourceTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::addDestinationTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::clearAnimationsForActiveWritingToolsSession):
(WebCore::ChromeClient::removeTransparentMarkersForSessionID): Deleted.
(WebCore::ChromeClient::removeInitialTextAnimation): Deleted.
(WebCore::ChromeClient::addInitialTextAnimation): Deleted.
(WebCore::ChromeClient::addSourceTextAnimation): Deleted.
(WebCore::ChromeClient::addDestinationTextAnimation): Deleted.
(WebCore::ChromeClient::clearAnimationsForSessionID): Deleted.
* Source/WebCore/page/Page.cpp:
(WebCore::Page::writingToolsSessionDidReceiveAction):
(WebCore::Page::contextRangeForActiveWritingToolsSession const):
(WebCore::Page::showSelectionForActiveWritingToolsSession const):
(WebCore::Page::contextRangeForSessionWithID const): Deleted.
(WebCore::Page::showSelectionForWritingToolsSessionWithID const): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/page/writing-tools/WritingToolsController.h:
* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::WritingToolsController::willBeginWritingToolsSession):
(WebCore::WritingToolsController::proofreadingSessionDidReceiveSuggestions):
(WebCore::WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion):
(WebCore::WritingToolsController::showSelection const):
(WebCore::WritingToolsController::compositionSessionDidFinishReplacement):
(WebCore::WritingToolsController::compositionSessionDidReceiveTextWithReplacementRangeAsync):
(WebCore::WritingToolsController::compositionSessionDidReceiveTextWithReplacementRange):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction&lt;WritingTools::Session::Type::Proofreading&gt;):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction&lt;WritingTools::Session::Type::Composition&gt;):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction):
(WebCore::WritingToolsController::didEndWritingToolsSession&lt;WritingTools::Session::Type::Proofreading&gt;):
(WebCore::WritingToolsController::didEndWritingToolsSession&lt;WritingTools::Session::Type::Composition&gt;):
(WebCore::WritingToolsController::didEndWritingToolsSession):
(WebCore::WritingToolsController::updateStateForSelectedSuggestionIfNeeded):
(WebCore::WritingToolsController::respondToUnappliedEditing):
(WebCore::WritingToolsController::respondToReappliedEditing):
(WebCore::WritingToolsController::activeSessionRange const):
(WebCore::WritingToolsController::currentState):
(WebCore::WritingToolsController::showOriginalCompositionForSession):
(WebCore::WritingToolsController::showRewrittenCompositionForSession):
(WebCore::WritingToolsController::restartCompositionForSession):
(WebCore::WritingToolsController::showSelectionForWritingToolsSessionWithID const): Deleted.
(WebCore::WritingToolsController::contextRangeForSessionWithID const): Deleted.
(WebCore::WritingToolsController::stateForSession): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView willBeginWritingToolsSession:requestContexts:]):
(-[WKWebView didEndWritingToolsSession:accepted:]):
(-[WKWebView _proofreadingSessionShowDetailsForSuggestionWithUUID:relativeToRect:]):
(-[WKWebView _proofreadingSessionUpdateState:forSuggestionWithUUID:]):
(-[WKWebView _enableSourceTextAnimationAfterElementWithID:]):
(-[WKWebView _enableFinalTextAnimationForElementWithID:]):
(-[WKWebView _proofreadingSessionWithUUID:showDetailsForSuggestionWithUUID:relativeToRect:]): Deleted.
(-[WKWebView _proofreadingSessionWithUUID:updateState:forSuggestionWithUUID:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::proofreadingSessionUpdateStateForSuggestionWithID):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPageProxy::enableTextAnimationTypeForElementWithID):
(WebKit::WebPageProxy::addTextAnimationForAnimationID):
(WebKit::WebPageProxy::showSelectionForActiveWritingToolsSession):
(WebKit::WebPageProxy::removeTextAnimationForAnimationID):
(WebKit::WebPageProxy::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPageProxy::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPageProxy::showSelectionForWritingToolsSessionAssociatedWithAnimationID): Deleted.
(WebKit::WebPageProxy::showSelectionForWritingToolsSessionWithID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/mac/WKTextAnimationManager.mm:
(-[WKTextAnimationManager addTextAnimationForAnimationID:withData:]):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebChromeClient::removeTransparentMarkersForActiveWritingToolsSession):
(WebKit::WebChromeClient::removeInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::addInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::addSourceTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::addDestinationTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::clearAnimationsForActiveWritingToolsSession):
(WebKit::WebChromeClient::removeTransparentMarkersForSessionID): Deleted.
(WebKit::WebChromeClient::removeInitialTextAnimation): Deleted.
(WebKit::WebChromeClient::addInitialTextAnimation): Deleted.
(WebKit::WebChromeClient::addSourceTextAnimation): Deleted.
(WebKit::WebChromeClient::addDestinationTextAnimation): Deleted.
(WebKit::WebChromeClient::clearAnimationsForSessionID): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm:
(WebKit::TextAnimationController::unreplacedRangeForActiveWritingToolsSession const):
(WebKit::TextAnimationController::contextRangeForActiveWritingToolsSession const):
(WebKit::TextAnimationController::contextRangeForTextAnimationID const):
(WebKit::TextAnimationController::removeTransparentMarkersForActiveWritingToolsSession):
(WebKit::TextAnimationController::removeInitialTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::addInitialTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::addSourceTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::addDestinationTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::clearAnimationsForActiveWritingToolsSession):
(WebKit::TextAnimationController::enableSourceTextAnimationAfterElementWithID):
(WebKit::TextAnimationController::enableTextAnimationTypeForElementWithID):
(WebKit::TextAnimationController::unreplacedRangeForSessionWithID const): Deleted.
(WebKit::TextAnimationController::contextRangeForSessionWithID const): Deleted.
(WebKit::TextAnimationController::removeTransparentMarkersForSessionID): Deleted.
(WebKit::TextAnimationController::removeInitialTextAnimation): Deleted.
(WebKit::TextAnimationController::addInitialTextAnimation): Deleted.
(WebKit::TextAnimationController::addSourceTextAnimation): Deleted.
(WebKit::TextAnimationController::addDestinationTextAnimation): Deleted.
(WebKit::TextAnimationController::showSelectionForWritingToolsSessionAssociatedWithAnimationID): Deleted.
(WebKit::TextAnimationController::clearAnimationsForSessionID): Deleted.
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::removeTransparentMarkersForActiveWritingToolsSession):
(WebKit::WebPage::removeInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::addInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::addSourceTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::addDestinationTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::clearAnimationsForActiveWritingToolsSession):
(WebKit::WebPage::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPage::enableTextAnimationTypeForElementWithID):
(WebKit::WebPage::showSelectionForActiveWritingToolsSession):
(WebKit::WebPage::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPage::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPage::removeTransparentMarkersForSessionID): Deleted.
(WebKit::WebPage::removeInitialTextAnimation): Deleted.
(WebKit::WebPage::addInitialTextAnimation): Deleted.
(WebKit::WebPage::addSourceTextAnimation): Deleted.
(WebKit::WebPage::addDestinationTextAnimation): Deleted.
(WebKit::WebPage::clearAnimationsForSessionID): Deleted.
(WebKit::WebPage::showSelectionForWritingToolsSessionAssociatedWithAnimationID): Deleted.
(WebKit::WebPage::showSelectionForWritingToolsSessionWithID): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: <a href="https://commits.webkit.org/282207@main">https://commits.webkit.org/282207@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fd011ad7feec8c4f453dcf88d76f18b421535c1d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62445 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15040 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/66429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/12997 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64565 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49487 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13333 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/66429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/12997 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65514 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/38846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54094 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/66429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/35549 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11407 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/11925 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/57186 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11723 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68158 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6391 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/11422 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/57695 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6420 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54118 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/57894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5316 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9397 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/37600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/38685 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/39782 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/38429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
```diff
@@ -249,7 +249,7 @@ struct PerWebProcessState {
 #if ENABLE(WRITING_TOOLS)
     RetainPtr<NSMapTable<NSUUID *, WTTextSuggestion *>> _writingToolsTextSuggestions;
-    RetainPtr<NSMapTable<NSUUID *, WTSession *>> _writingToolsSessions;
+    WTSession *_activeWritingToolsSession;
```
oh maybe this should be RetainPtr?

rr-codes commented on Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
```diff
@@ -1223,26 +1223,24 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
 
 void WebPageProxy::addTextAnimationForAnimationID(IPC::Connection& connection, const WTF::UUID& uuid, const WebCore::TextAnimationData& styleData, const WebCore::TextIndicatorData& indicatorData, CompletionHandler<void(WebCore::TextAnimationRunMode)>&& completionHandler)
 {
-    MESSAGE_CHECK(uuid.isValid());
```
put this back

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm:
```diff
@@ -172,10 +165,10 @@
 {
-    if (auto animationID = m_initialAnimations.take(sessionID))
-        m_webPage->removeTextAnimationForAnimationID(animationID);
+    if (auto initialAnimationID = m_initialAnimationID)
```
I guess this also needs to set m_initialAnimationID to nullopt

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm:
```diff
@@ -113,49 +112,43 @@
+        removeTransparentMarkersForTextAnimationID(*initialAnimationID);
 
-    auto animationStates = m_textAnimationRanges.take(sessionID);
+    auto animationStates = m_textAnimationRanges;
```
this should set m_textAnimationRanges to { } i think

megangardner replied:
I think yes, that you can clear/reset basically everything in the function. Perhaps it should be renamed as such.

rr-codes commented on Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm:
```diff
@@ -388,19 +347,16 @@
-        for (auto animationRange : animationRanges)
-            m_webPage->removeTextAnimationForAnimationID(animationRange.animationUUID);
-    }
+    for (auto animationRange : m_textAnimationRanges)
```
this should reset m_textAnimationRanges back to { }

pxlcoder commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -227,26 +225,26 @@
+        return;
+    }
+
+    RELEASE_LOG(WritingTools, "WritingToolsController::proofreadingSessionDidReceiveSuggestions (%s) [received suggestions: %zu, finished: %d]", state->session.identifier.toString().utf8().data(), suggestions.size(), finished);
```
How about moving the log back to the top of the method, and another log statement in the early return?

pxlcoder commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -286,18 +284,18 @@
 
-    CheckedPtr state = stateForSession<WritingTools::Session::Type::Proofreading>(session);
-    if (!state) {
+    RELEASE_LOG(WritingTools, "WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion (%s) [new state: %hhu, suggestion: %s]", state->session.identifier.toString().utf8().data(), enumToUnderlyingType(newTextSuggestionState), textSuggestion.identifier.toString().utf8().data());
```
Same comment about logging.

pxlcoder commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -490,40 +482,46 @@
+    }
 
-    m_page->chrome().client().removeInitialTextAnimation(session.identifier);
+    RELEASE_LOG(WritingTools, "WritingToolsController::compositionSessionDidReceiveTextWithReplacementRange (%s) [range: %llu, %llu; finished: %d]", state->session.identifier.toString().utf8().data(), range.location, range.length, finished);
```
Same comment about logging.

pxlcoder commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -490,40 +482,46 @@
 
-    CheckedPtr state = stateForSession<WritingTools::Session::Type::Proofreading>(session);
-    if (!state) {
+    RELEASE_LOG(WritingTools, "WritingToolsController::writingToolsSessionDidReceiveAction<Proofreading> (%s) [action: %hhu]", state->session.identifier.toString().utf8().data(), enumToUnderlyingType(action));
```
Same comment about logging.

pxlcoder commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -569,51 +567,63 @@
+        return;
+    }
+
+    RELEASE_LOG(WritingTools, "WritingToolsController::writingToolsSessionDidReceiveAction<Composition> (%s) [action: %hhu]", state->session.identifier.toString().utf8().data(), enumToUnderlyingType(action));
```
Same comment about logging.

pxlcoder commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -569,51 +567,63 @@
+    }
 
-    switch (session.type) {
+    RELEASE_LOG(WritingTools, "WritingToolsController::writingToolsSessionDidReceiveAction (%s) [action: %hhu]", state->session.identifier.toString().utf8().data(), enumToUnderlyingType(action));
```
Same comment about logging.

pxlcoder commented on Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
```diff
@@ -1311,27 +1303,21 @@ static bool exceedsRenderTreeSizeSizeThreshold(uint64_t thresholdSize, uint64_t
 
 void WebPageProxy::removeTextAnimationForAnimationID(IPC::Connection& connection, const WTF::UUID& uuid)
 {
-    MESSAGE_CHECK(uuid.isValid());
```
Why is this being removed?

pxlcoder commented on Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm:
```diff
@@ -172,10 +166,12 @@
+    if (auto initialAnimationID = m_initialAnimationID)
+        m_webPage->removeTextAnimationForAnimationID(*initialAnimationID);
+
+    m_initialAnimationID = std::nullopt;
```
You can use `std::exchange` here.

rr-codes committed:
[Writing Tools] Remove unnecessary infrastructure to support concurrent Writing Tools sessions
https://bugs.webkit.org/show_bug.cgi?id=277998
rdar://133726884

Reviewed by Aditya Keerthi.

There's no need to support multiple sessions, and doing so significantly increases the complexity of the code.

* Source/WebCore/dom/DocumentMarker.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebCore::ChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebCore::ChromeClient::removeTransparentMarkersForActiveWritingToolsSession):
(WebCore::ChromeClient::removeInitialTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::addInitialTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::addSourceTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::addDestinationTextAnimationForActiveWritingToolsSession):
(WebCore::ChromeClient::clearAnimationsForActiveWritingToolsSession):
(WebCore::ChromeClient::removeTransparentMarkersForSessionID): Deleted.
(WebCore::ChromeClient::removeInitialTextAnimation): Deleted.
(WebCore::ChromeClient::addInitialTextAnimation): Deleted.
(WebCore::ChromeClient::addSourceTextAnimation): Deleted.
(WebCore::ChromeClient::addDestinationTextAnimation): Deleted.
(WebCore::ChromeClient::clearAnimationsForSessionID): Deleted.
* Source/WebCore/page/Page.cpp:
(WebCore::Page::writingToolsSessionDidReceiveAction):
(WebCore::Page::contextRangeForActiveWritingToolsSession const):
(WebCore::Page::showSelectionForActiveWritingToolsSession const):
(WebCore::Page::contextRangeForSessionWithID const): Deleted.
(WebCore::Page::showSelectionForWritingToolsSessionWithID const): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/page/writing-tools/WritingToolsController.h:
* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::WritingToolsController::willBeginWritingToolsSession):
(WebCore::WritingToolsController::proofreadingSessionDidReceiveSuggestions):
(WebCore::WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion):
(WebCore::WritingToolsController::showSelection const):
(WebCore::WritingToolsController::compositionSessionDidFinishReplacement):
(WebCore::WritingToolsController::compositionSessionDidReceiveTextWithReplacementRangeAsync):
(WebCore::WritingToolsController::compositionSessionDidReceiveTextWithReplacementRange):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction<WritingTools::Session::Type::Proofreading>):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction<WritingTools::Session::Type::Composition>):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction):
(WebCore::WritingToolsController::didEndWritingToolsSession<WritingTools::Session::Type::Proofreading>):
(WebCore::WritingToolsController::didEndWritingToolsSession<WritingTools::Session::Type::Composition>):
(WebCore::WritingToolsController::didEndWritingToolsSession):
(WebCore::WritingToolsController::updateStateForSelectedSuggestionIfNeeded):
(WebCore::WritingToolsController::respondToUnappliedEditing):
(WebCore::WritingToolsController::respondToReappliedEditing):
(WebCore::WritingToolsController::activeSessionRange const):
(WebCore::WritingToolsController::currentState):
(WebCore::WritingToolsController::showOriginalCompositionForSession):
(WebCore::WritingToolsController::showRewrittenCompositionForSession):
(WebCore::WritingToolsController::restartCompositionForSession):
(WebCore::WritingToolsController::showSelectionForWritingToolsSessionWithID const): Deleted.
(WebCore::WritingToolsController::contextRangeForSessionWithID const): Deleted.
(WebCore::WritingToolsController::stateForSession): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView willBeginWritingToolsSession:requestContexts:]):
(-[WKWebView didEndWritingToolsSession:accepted:]):
(-[WKWebView _proofreadingSessionShowDetailsForSuggestionWithUUID:relativeToRect:]):
(-[WKWebView _proofreadingSessionUpdateState:forSuggestionWithUUID:]):
(-[WKWebView _enableSourceTextAnimationAfterElementWithID:]):
(-[WKWebView _enableFinalTextAnimationForElementWithID:]):
(-[WKWebView _proofreadingSessionWithUUID:showDetailsForSuggestionWithUUID:relativeToRect:]): Deleted.
(-[WKWebView _proofreadingSessionWithUUID:updateState:forSuggestionWithUUID:]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::proofreadingSessionUpdateStateForSuggestionWithID):
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPageProxy::enableTextAnimationTypeForElementWithID):
(WebKit::WebPageProxy::addTextAnimationForAnimationID):
(WebKit::WebPageProxy::showSelectionForActiveWritingToolsSession):
(WebKit::WebPageProxy::removeTextAnimationForAnimationID):
(WebKit::WebPageProxy::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPageProxy::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPageProxy::showSelectionForWritingToolsSessionAssociatedWithAnimationID): Deleted.
(WebKit::WebPageProxy::showSelectionForWritingToolsSessionWithID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/mac/WKTextAnimationManager.mm:
(-[WKTextAnimationManager addTextAnimationForAnimationID:withData:]):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebChromeClient::removeTransparentMarkersForActiveWritingToolsSession):
(WebKit::WebChromeClient::removeInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::addInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::addSourceTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::addDestinationTextAnimationForActiveWritingToolsSession):
(WebKit::WebChromeClient::clearAnimationsForActiveWritingToolsSession):
(WebKit::WebChromeClient::removeTransparentMarkersForSessionID): Deleted.
(WebKit::WebChromeClient::removeInitialTextAnimation): Deleted.
(WebKit::WebChromeClient::addInitialTextAnimation): Deleted.
(WebKit::WebChromeClient::addSourceTextAnimation): Deleted.
(WebKit::WebChromeClient::addDestinationTextAnimation): Deleted.
(WebKit::WebChromeClient::clearAnimationsForSessionID): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.h:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm:
(WebKit::TextAnimationController::unreplacedRangeForActiveWritingToolsSession const):
(WebKit::TextAnimationController::contextRangeForActiveWritingToolsSession const):
(WebKit::TextAnimationController::contextRangeForTextAnimationID const):
(WebKit::TextAnimationController::removeTransparentMarkersForActiveWritingToolsSession):
(WebKit::TextAnimationController::removeInitialTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::addInitialTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::addSourceTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::addDestinationTextAnimationForActiveWritingToolsSession):
(WebKit::TextAnimationController::clearAnimationsForActiveWritingToolsSession):
(WebKit::TextAnimationController::enableSourceTextAnimationAfterElementWithID):
(WebKit::TextAnimationController::enableTextAnimationTypeForElementWithID):
(WebKit::TextAnimationController::unreplacedRangeForSessionWithID const): Deleted.
(WebKit::TextAnimationController::contextRangeForSessionWithID const): Deleted.
(WebKit::TextAnimationController::removeTransparentMarkersForSessionID): Deleted.
(WebKit::TextAnimationController::removeInitialTextAnimation): Deleted.
(WebKit::TextAnimationController::addInitialTextAnimation): Deleted.
(WebKit::TextAnimationController::addSourceTextAnimation): Deleted.
(WebKit::TextAnimationController::addDestinationTextAnimation): Deleted.
(WebKit::TextAnimationController::showSelectionForWritingToolsSessionAssociatedWithAnimationID): Deleted.
(WebKit::TextAnimationController::clearAnimationsForSessionID): Deleted.
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::removeTransparentMarkersForActiveWritingToolsSession):
(WebKit::WebPage::removeInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::addInitialTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::addSourceTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::addDestinationTextAnimationForActiveWritingToolsSession):
(WebKit::WebPage::clearAnimationsForActiveWritingToolsSession):
(WebKit::WebPage::enableSourceTextAnimationAfterElementWithID):
(WebKit::WebPage::enableTextAnimationTypeForElementWithID):
(WebKit::WebPage::showSelectionForActiveWritingToolsSession):
(WebKit::WebPage::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPage::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPage::removeTransparentMarkersForSessionID): Deleted.
(WebKit::WebPage::removeInitialTextAnimation): Deleted.
(WebKit::WebPage::addInitialTextAnimation): Deleted.
(WebKit::WebPage::addSourceTextAnimation): Deleted.
(WebKit::WebPage::addDestinationTextAnimation): Deleted.
(WebKit::WebPage::clearAnimationsForSessionID): Deleted.
(WebKit::WebPage::showSelectionForWritingToolsSessionAssociatedWithAnimationID): Deleted.
(WebKit::WebPage::showSelectionForWritingToolsSessionWithID): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:

Canonical link: https://commits.webkit.org/282207@main
